src/allmydata/frontends/magic_folder.py

```diff
@@ -1053,6 +1064,22 @@ def _process_deque(self):
 
                 self._call_hook(proc, 'processed')
 
+    def _trim_process_history(self):
+        """
+        Remove any history items which are 'too old' (which is currently
+        defined as more than 30s).
+        """
+        now = time.time()
+        # print("trim history: now={}".format(len(self._process_history)))
+        # for item in self._process_history:
+        #     print("  {}: {}".format(item.relpath_u, (now - item.latest_update())))
+        self._process_history = [
+            item
+            for item in self._process_history
+            if not item.latest_update() or (now - item.latest_update()) < 30.0
+        ]
```

Comment:
`latest_update() is None` rather than `not latest_update()`.

Also, I think this definition of process history is pretty complex and will probably be pretty confusing.

First, `_process_history` documentation is wrong.  It says it contains "completed" items but it really contains "started and completed items" since items are added to it immediately before `_process` and only removed by this trim function (there is one call to `_process_history.remove` elsewhere but I think it is impossible to reach...).

This seems to be part of why items can appear twice in magic-folder status.  They can be in progress and be in both `_in_progress` and `_process_history` (not new in this PR).  That's worth fixing separately, I guess.  I think it's something that's been complained about from the GridSync side.

Once that's fixed then `latest_update` is probably not necessary: in-progress items are the items that are receiving updates and they are in `_in_progress`; completed items have some single immutable completion time and, most likely, they will be appended to `_process_history` so that the list is monotonically increasing by completion time.  In that case, trimming it just involves dropping some items from the front instead of scanning the whole thing and looking at update times.

Reply:
Additionally, making the amount of time to trim for a parameter somewhere is probably useful.